### PR TITLE
feat: add-genesis-oracle CLI

### LINKED ISSUE #531: Implement initial Oracle Registration
- add initial oracle with governance proposal
- set `Params` oracle pubkey, remote report and unique id
- add info to `Oracle` and `OracleRegistration`

### PR DESCRIPTION
**Pull request by H4NLee**
close #531 

Implementation of `add-genesis-oracle` CLI.

### CODE REVIEW

**Review by gyuguen — APPROVED** [review]
LGTM

File: cmd/panacead/cmd/genoracle.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		Use:   "add-genesis-oracle",
+		Short: "Add a genesis oracle and oracle module parameters to genesis.json",
+		Long: `
+			Using this command, you can set genesis oracle and oracle module parameters.
+			If you set oracle public key path for oracle module params, the flags for oracle public key and remote report would be ignored.
+
+			The desired format of oracle public key file is:
+
+			"public_key_base64" : <base64-encoded-oracle-public-key>
+			"remote_report_base64" : <base64-encoded-oracle-remote-report>
```
Comment (youngjoon-lee) [inline cmd/panacead/cmd/genoracle.go:49]: We need to mention that we expect a JSON file.
```suggestion
			The desired format of oracle public key file is:

			{
				"public_key_base64" : "<base64-encoded-oracle-public-key>",
				"remote_report_base64" : "<base64-encoded-oracle-remote-report>"
			}
```

File: cmd/panacead/cmd/genoracle.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+func setOracle(cmd *cobra.Command, genState *oracletypes.GenesisState) error {
+	clientCtx := client.GetClientContextFromCmd(cmd)
+
+	oracleAddressOrKey, err := cmd.Flags().GetString(flagOracleAccount)
+	if err != nil {
+		return fmt.Errorf("failed to get oracle account flag: %w", err)
+	}
+
+	if len(oracleAddressOrKey) > 0 {
```
Comment (youngjoon-lee) [inline cmd/panacead/cmd/genoracle.go:116]: This code would be much readable if you escape this function asap.
```suggestion
	if len(oracleAddressOrKey) == 0 {
	    return
	}
```
, or returning an error. I'm not sure which way is correct.

**Review by youngjoon-lee — APPROVED** [review]
lgtm

File: cmd/panacead/cmd/genoracle.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+func setOracle(cmd *cobra.Command, genState *oracletypes.GenesisState) error {
+	clientCtx := client.GetClientContextFromCmd(cmd)
+
+	oracleAddressOrKey, err := cmd.Flags().GetString(flagOracleAccount)
+	if err != nil {
+		return fmt.Errorf("failed to get oracle account flag: %w", err)
+	}
+
+	if len(oracleAddressOrKey) == 0 {
+		return nil
```
Comment (inchori) [inline cmd/panacead/cmd/genoracle.go:118]: I think that it need to return error.

**Review by inchori — APPROVED** [review]
lgtm!
I left a one comment.

File: cmd/panacead/cmd/genoracle.go (hunk 3 above)
Comment (H4NLee) [reply cmd/panacead/cmd/genoracle.go]: fixed in [7ed8078](https://github.com/medibloc/panacea-core/pull/536/commits/7ed80785619c033980de5153154ddc4abd686b0e)

**Comment by H4NLee** [comment]
> It seems that it can be used for adding genesis Oracle information for testing. However, it seems that we need to use the upgrade handler to add Oracle in the actual running panacea. (Or make proposal for it)

Right. We need to make a proposal to register a genesis oracle in mainnet.

**Pull request merged**